crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -218,26 +223,31 @@ impl ShaderCache {
 
 #[derive(Default)]
 struct LayoutCache {
-    layouts: HashMap<Vec<BindGroupLayoutId>, wgpu::PipelineLayout>,
+    layouts: HashMap<(Vec<BindGroupLayoutId>, Vec<PushConstantRange>), wgpu::PipelineLayout>,
```

Comment:
This key is getting pretty darn big. We should probably consider making this a dedicated type and use `bevy_utils::PreHashMap`

Reply:
@james7132 I don't understand how `PreHashMap` helps in this case. `Hashed<K>` also stores `K`. If anything, using `PreHashMap` makes the key **bigger** because it memoizes the `u64` hash.

I'm necromancing this thread b/c @Neo-Zhixing is trying to do something similar in #7681 .